File: Programming/Practice/24/C++/24/24/24.cpp
#include <iostream>
#include <fstream>
#include <string>
#include <iomanip>
#include "json.h"

using namespace std;
using json = nlohmann::json;

int main()
{
	json inlocal; // Для работы с in.json
	json outlocal = json::array(); // Для работы с out.json


	ifstream fin("in.json");
	fin >> inlocal;

	int usr_cmplt = 0;// Кол-во юзеров 
	int completed;// Кол-во заданий сделанных одним юзером

	for(int i = 0; i < inlocal.size(); i++)// Цикл считает кол-во юзеров
	{
		if (usr_cmplt < inlocal[i]["userId"])
		{
			usr_cmplt++;
		}

	}
	for (int i = 1; i <= usr_cmplt; i++)// i = 1, а не i = 0, потому что нет юзера под номером 0
	{
		completed = 0;// В каждой итерации значение обнуляется 

		for(int j = 0; j < inlocal.size(); j++)// Цикл считает сколько верно решенных задач у юзера под номером i
		{
			if ((inlocal[j]["completed"] == true) && (inlocal[j]["userId"] == i))// Проверяется выполнено ли задание(completed) и выполнено ли оно одни и тем же пользователем(userId) 
			{
				completed++;
			}
		}

		if ((completed >= 1) && (completed <= 1000000000))// Требование из условия задачи
		{
			json data_usr;
			data_usr["task_completed"] = completed;
			data_usr["userId"] = i;
			outlocal.push_back(data_usr);// Создается json data_usr с ключами "task_completed" и "userId" и им присваивается соответствующее значение и вписывается в outlocal
		}

	}
	fin.close();// Закрытие файла
	ofstream fout("out.json");
	fout << std::setw(2) << outlocal << std::endl;// Перенос содержимого из utlocal в out.json
	fout.close();// Закрытие файла
}

File: Programming/Practice/24/Python/24.py
import json
import os

usr_cmplt = 0 # Кол-во пользователей
with open ('in.json', 'r', encoding = 'utf-8') as read_in:
	inlocal = json.load(read_in)# В inlocal заптсывается содержимое in.json
outlocal = []
# Цикл, который вычисляет кол-во пользователей:
for i in inlocal: 
	if int(usr_cmplt) < int(i["userId"]):
		usr_cmplt += 1

for j in range(usr_cmplt + 1):#usr_cmplt + 1, потому что отсчет идет с 0
	completed = 0
	# Внутренний цикл считает сколько заданий сделал один пользователь:
	for n in range(len(inlocal)):
		if inlocal[n]["completed"] == True and inlocal[n]["userId"] == j:
			completed += 1
	# Запись данных о пользователе в outlocal:
	if completed >= 1 and completed <= 1000000000: # Условие задачи
		data = { "task_completed":completed, "userID":j}
		outlocal += [data]
# Запись содержимого outlocal в out.json:
with open("out.json", "w") as write_out:
    json.dump(outlocal, write_out, indent=2)